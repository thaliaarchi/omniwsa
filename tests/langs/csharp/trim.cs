using System;

class Program {
    static void Main() {
        for (int codeUnit = 0; codeUnit <= 0xFFFF; codeUnit++) {
            char c = (char)codeUnit;
            string both = $"{c}x{c}";
            string trimmed = both.Trim();
            if (trimmed != both) {
                string direction;
                if (trimmed == "x") {
                    direction = "on both sides";
                } else if (trimmed == $"x{c}") {
                    direction = "only on the left";
                } else if (trimmed == $"{c}x") {
                    direction = "only on the right";
                } else {
                    direction = "BUG";
                }
                Console.WriteLine($"U+{codeUnit:X4} is stripped {direction}");
            }
        }
    }
}
