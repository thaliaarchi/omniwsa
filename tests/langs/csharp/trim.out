Mono C# compiler version 6.12.0.0
U+0009 is stripped on both sides
U+000A is stripped on both sides
U+000B is stripped on both sides
U+000C is stripped on both sides
U+000D is stripped on both sides
U+0020 is stripped on both sides
U+0085 is stripped on both sides
U+00A0 is stripped on both sides
U+1680 is stripped on both sides
U+2000 is stripped on both sides
U+2001 is stripped on both sides
U+2002 is stripped on both sides
U+2003 is stripped on both sides
U+2004 is stripped on both sides
U+2005 is stripped on both sides
U+2006 is stripped on both sides
U+2007 is stripped on both sides
U+2008 is stripped on both sides
U+2009 is stripped on both sides
U+200A is stripped on both sides
U+2028 is stripped on both sides
U+2029 is stripped on both sides
U+202F is stripped on both sides
U+205F is stripped on both sides
U+3000 is stripped on both sides
